### ISSUE TITLE
Avoid more occurrences of the term 'sidebar' in user-facing strings

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#61499]
### Description

In https://github.com/WordPress/gutenberg/issues/49613 and https://github.com/WordPress/gutenberg/pull/49614 the term 'sidebar' was removed from most of the user-facing text / labels / names.

There is also an issue open to remove the term 'sidebar' from the Handbook, see https://github.com/WordPress/gutenberg/issues/57546

In the meantime. I noticed that new occurrences of the term 'sidebar' are still in the codebase and they should be replaced by a more generic, non-biased, term such as 'panel'.

To recap the main reasons why the term 'sidebar' should be avoided:
- The whole concept of 'sidebar' is extraneous to blind screen reader users. The term 'sidebar' is a positional reference, as it refers to something that is placed on a 'side'. As a blind user, it doesn't help me to know whether a section of the user interface is visually placed on the side, top, bottom or in any other spot of the screen.
- Ideally, UI sections should be named based on their functionality rather than on their visual aspect. Same applies for components.
- On small screens, most of these UI sections aren't 'sidebars' any longer. They're more full-width overlay panels so that the term 'sidebar' isn't accurate in this case.
- When translated, the term 'sidebar' is generally way longer than in English and that doesn't help readability.

It would be great to educate all contributors to not use the term 'sidebar' other than for internal purposes. It should never be used in user-facing strings or public documentation.

### Step-by-step reproduction instructions

Occurrences found in the codebase as of May 2024:

- `_e( 'You can change the amount of columns in your galleries by dragging a slider in the block inspector in the sidebar.', 'gutenberg' );`
- `label: __( 'Toggle settings sidebar' )` - in the commands palette, common commands 
- `label: __( 'Toggle settings sidebar' )` - in the commands palette, edit mode commands 
- `el( PanelBody, {}, __( 'My sidebar content' ) )` in the editor README, + 3 more occurrences
- `description: __( 'Show or hide the Settings sidebar.' ),` in the global keyboard shortcuts
- a few occurrences in the [plugin sidebar](https://github.com/WordPress/gutenberg/blob/14ecb1d627335fa5b1f9b6999fb9f4f2fe6127cd/packages/editor/src/components/plugin-sidebar/index.js#L52) docblock
- a few occurrences in the [plugin sidebar more menu item](https://github.com/WordPress/gutenberg/blob/14ecb1d627335fa5b1f9b6999fb9f4f2fe6127cd/packages/editor/src/components/plugin-sidebar-more-menu-item/index.js) docblock

Note; while the occurrences in the docblocks aren't technically user-facing, they do suggest to use the term 'sidebar' in user-facing strings. As such, they should be changes as well.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by n2erjo00** [comment]
@afercia There is new shiny PR which replaces sidebar usage

**Comment by im3dabasia** [comment]
@afercia, thank you for raising the issue.

I worked on a similar PR that dealt with avoiding/removing the word "toggle" in user-facing strings. Here is the link to the [PR](https://github.com/WordPress/gutenberg/pull/67741). Since these are related changes, in this issue we are also trying to avoid more occurrences of the term "sidebar." Do you think it would make sense to add a `no-restricted-syntax` rule for this in the `.eslintrc.js` to prevent the usage of this word in translatable strings?

Looking forward to hearing your thoughts.

**Comment by afercia** [comment]
@im3dabasia thanks for your feedback.
Ideally, I would like documentation and education of designers and contributors to be enough to understand the problem and avoid the term 'sidebar'. That doesn't seem sufficient though so I'd encourage to explore your suggestion and introduce a linting rule.

The rule for `toggle` added in https://github.com/WordPress/gutenberg/pull/67741 is great and modern IDEs do show linting violation while editing, which is great. Screenshot from Visual Studio Code:

![Image](https://github.com/user-attachments/assets/8ec0a457-e6d6-4833-b933-0efc112ac6a0)

I'm totally in favor for a similar rule for the term 'sidebar'.
To recap why it should be avoided:

- 'Sidebar' is a spatial concept as it refers to a 'side'. Non sighted users have no benefit from a message that refers to spatial positioning. It's also slightly unfair, as it would be referring to colors for example: "Click the red button" isn't fair for blind users. Similarly, 'Click the button in the right sidebar" isn't fair. A more appropriate wording would be "in the settings panel".
- Most of these 'sidebars' in the mobile view aren't 'sidebars' any longer. Rather, they open like a full-width overlay.
